pkg/cnab/extensions/solver.go

```diff
@@ -42,16 +45,57 @@ func (s *DependencySolver) ResolveDependencies(bun *bundle.Bundle) ([]Dependency
 	return q, nil
 }
 
-func (s *DependencySolver) ResolveVersion(name string, dep Dependency) (string, error) {
+func (s *DependencySolver) ResolveVersion(alias string, dep Dependency) (string, error) {
 	// Here is where we could split out this logic into multiple strategy funcs / structs if necessary
-	if dep.Version == nil {
+	if dep.Version == nil || len(dep.Version.Ranges) == 0 {
 		parts := strings.Split(dep.Bundle, ":")
 		if len(parts) > 1 {
 			return strings.Join(parts[1:], ""), nil
 		} else {
-			return "", errors.Errorf("not implemented: unspecified dependency version for %s", name)
+			return s.determineDefaultTag(dep)
 		}
 	}
 
-	return "", errors.Errorf("not implemented: dependency version range specified for %s", name)
+	return "", errors.Errorf("not implemented: dependency version range specified for %s", alias)
+}
+
+func (s *DependencySolver) determineDefaultTag(dep Dependency) (string, error) {
+	tags, err := crane.ListTags(dep.Bundle)
```

Comment:
crane ftw!

Reply:
Yes it is doing a lot of heavy lifting under the hood. You have no idea!